osu.Game.Rulesets.Taiko/Difficulty/Evaluators/ColourEvaluator.cs

```diff
@@ -36,18 +37,64 @@ public static double EvaluateDifficultyOf(RepeatingHitPatterns repeatingHitPatte
             return 2 * (1 - DifficultyCalculationUtils.Logistic(exponent: Math.E * repeatingHitPattern.RepetitionInterval - 2 * Math.E));
         }
 
+        /// <summary>
+        /// Calculates a consistency penalty based on the number of consecutive consistent intervals,
+        /// considering the delta time between each colour sequence.
+        /// </summary>
+        /// <param name="hitObject">The current hitObject to consider.</param>
+        /// <param name="threshold"> The allowable margin of error for determining whether ratios are consistent.</param>
+        private static double consistentRatioPenalty(TaikoDifficultyHitObject hitObject, double threshold = 0.01)
+        {
+            int consistentRatioCount = 0;
+            double totalRatioCount = 0.0;
+
+            TaikoDifficultyHitObject current = hitObject;
+
+            while (current.Previous(1) is TaikoDifficultyHitObject previousHitObject)
```

Comment:
This feels really iffy. Did you test it on some absurdly long maps? Since diffcalc might run just when scrolling through the  song select this might noticeably lag people's game, and when taking pp counter in account (which runs diffcalc pretty much every object) I think this might be unacceptable.
Wouldn't using a constant amount of objects be enough here? Even something big like 256 would be way better